packages/lodestar/src/chain/lightClient/index.ts

```diff
@@ -468,23 +468,36 @@ export class LightClientServer {
       return;
     }
 
-    const newPartialUpdate: PartialLightClientUpdate = attestedData.isFinalized
-      ? {
-          ...attestedData,
-          finalizedHeader: await this.getFinalizedHeader(attestedData.finalizedCheckpoint.root as Uint8Array),
-          syncAggregate,
-        }
-      : {
-          ...attestedData,
-          syncAggregate,
-        };
+    let newPartialUpdate: PartialLightClientUpdate;
+
+    if (attestedData.isFinalized) {
+      // If update if finalized retrieve the previously stored header from DB.
+      // Only checkpoint candidates are stored, and not all headers are guaranteed to be available
+      const finalizedCheckpointRoot = attestedData.finalizedCheckpoint.root as Uint8Array;
+      const finalizedHeader = await this.getFinalizedHeader(finalizedCheckpointRoot);
+      if (finalizedHeader) {
+        // If finalizedHeader is available (should be most times) create a finalized update
+        newPartialUpdate = {...attestedData, finalizedHeader, syncAggregate};
+      } else {
+        // If finalizedHeader is not available (happens on startup) create a non-finalized update
+        newPartialUpdate = {...attestedData, isFinalized: false, syncAggregate};
```

Comment:
if it's not at startup time, is it an issue of missing finalizedHeader? Should we add new metric to track it `missingFinalizedHeader`?

Reply:
Just added a metric to track that in https://github.com/ChainSafe/lodestar/pull/3771/commits/9ef5e36f6dfa476a68b2b07db320b44ee9548f77

Yes it can be a problem if it's not available latter, but it's a situation that can happen depending on network conditions. We should monitor this new metric from now on